crossbuilder

```diff
@@ -145,15 +145,25 @@ variables () {
     SOURCE_PATH_LOCAL=$MOUNTED_DIRECTORY
     SOURCE_PATH_CONTAINER=$MOUNT_POINT
     POST_DEPLOY_SCRIPT=.crossbuilder/post_deploy
-    HOST_FARCH=$(dpkg-architecture -f -a$HOST_ARCH -qDEB_BUILD_MULTIARCH)
-    TARGET_FARCH=$(dpkg-architecture -f -a$TARGET_ARCH -qDEB_HOST_MULTIARCH)
-    if [ "$(dpkg-architecture -f -a$TARGET_ARCH -qDEB_BUILD_ARCH)" != "$TARGET_ARCH" ]; then
-        DEB_BUILD_PROFILES="cross nocheck nodoc noudeb"
-        EXTRA_DEB_BUILD_OPTIONS="nocheck nodoc noudeb"
-    else
-        DEB_BUILD_PROFILES="noudeb"
-        EXTRA_DEB_BUILD_OPTIONS="noudeb"
+
+    # To fill the HOST_FARCH, TARGET_FARCH, DEB_BUILD_PROFILES and EXTRA_DEB_BUILD_OPTIONS variables,
+    # you need the `dpkg-architecture' command, which is not always installed on host macine,
+    # but is present in the ubports sdk container.
+    # So if there is no 'dpkg-architecture' command, don't worry, the variables will be filed later, upon container is ensured.
+    if which dpkg-architecture > /dev/null 2>&1 ; then
+        HOST_FARCH=$(dpkg-architecture -f -a$HOST_ARCH -qDEB_BUILD_MULTIARCH)
```

Comment:
No,there is no need for the ```dpkg-architecture``` to be executed on the host. It can wait until container creation/ensurement. I could remove the package detection and usage on the host and move it entirely to container. Will it be OK?

Reply:
Done. I've also deleted all the unnecessary ```detect_farch_and_build_params_in_container_if_needed``` calls.